### PR TITLE
fix / Support latest PolkaDEX API

### PR DESCRIPTION
**Pull request by dizpers**
**Before submitting this PR, please make sure**:

- [ ] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

PolkaDEX team had updated their APIs (with breaking changes), so the connector in public version of Hummingbot became broken. This PR made all the fixes needed for this connector to work properly and also include some improvements (like result pagination).

The test suite still need to be updated, so I'm disabling it for now.

**Tests performed by the developer**:

1. Manually tested in local environment (following [HF checklist](https://hummingbot.org/developers/connectors/test/))
2. Bots are successfully running in production environment.

**Tips for QA testing**:

N/A

### CODE REVIEW

**Comment by nikspz** [comment]
@dizpers thank you for submitting PR, fyi, it should be aimed to development, I changed it for now https://hummingbot.org/developers/contributions/#checklist

**Comment by rapcmia** [comment]
Hi @dizpers good day
Can you take a look on the branch conflicts? thanks

**Comment by dizpers** [comment]
New API docs: https://app.theneo.io/polkadex/orderbook-api-docs/api-reference

**Comment by dizpers** [comment]
@rapcmia @nikspz conflicts are resolved now

**Comment by dizpers** [comment]
Looks like it requires tests to be fixed as well

**Review by rapcmia — APPROVED** [review]
LGTM
- Ran tests on source build local macos
- Create new account and use for setting up the connector, all ok 
- Balance displayed ok 
- Using v1 strategy (pmm)
    - Orders are created, fetched and cancelled successfully
    - Trade orders are recorded matched between history, CSV and exchange trade history
    - Run for couple of hours with low and high refresh time ok
- Build docker image ok

**Comment by dizpers** [comment]
@rapcmia All PolkaDEX tests has been fixed. So I'm enabling them back in Makefile and in coverage configuration.

**Comment by Gauthamastro** [comment]
Hey guys, is there anything blocking this to be merged? Can we help somehow to merge this sooner?

**Comment by nikspz** [comment]
hi @Gauthamastro Yes, we need to add test coverage here 

@dizpers 
https://github.com/hummingbot/hummingbot/actions/runs/9007313287/job/24764466360?pr=7016

**Comment by dizpers** [comment]
@Gauthamastro @nikspz yes, Enflux team is working on that now.

**Comment by dizpers** [comment]
@rapcmia @nikspz coverage should be 80%+ now, could your run the workflow?

**Comment by rapcmia** [comment]
Hi @dizpers thanks for the update, the github checks are failing on some tests below:
image

**Comment by dizpers** [comment]
@rapcmia @nikspz could you run the pipeline one more time?

File: hummingbot/connector/exchange/polkadex/polkadex_constants.py [hunk 1]
```diff
@@ -76,8 +80,8 @@
     ),
     RateLimit(
         limit_id=PLACE_ORDER_LIMIT_ID,
-        limit=NO_LIMIT,
-        time_interval=SECOND,
+        limit=2,
```
Comment (cardosofede) [inline hummingbot/connector/exchange/polkadex/polkadex_constants.py:83]: only supports 2 orders every 2 secs?

**Review by cardosofede — APPROVED** [review]
LGTM!

**Comment by rapcmia** [comment]
Merged to development and part of release version 1.28.0

**Pull request merged**